### PR TITLE
171025 Project Skeleton Creation Mock Api

### PR DESCRIPTION
**Pull request by shirinbt14926**
- Project Skeleton for API, Wiremock Stub and Unit Tests.
- Endpoints for Stub

This PBI includes  ,PBI-171032
- added dependency checker, unit test and coverage tasks
- added build and publish project artifacts tasks

### CODE REVIEW

File: src/UKHO.S100PermitService.Stubs/Program.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using UKHO.S100PermitService.Stubs.Configuration;
+using WireMock.Server;
+using WireMock.Settings;
+
+namespace UKHO.S100PermitService.Stubs
+{
+    class Program
```
Comment (Nadeem14456) [inline src/UKHO.S100PermitService.Stubs/Program.cs:10]:
```suggestion
    internal static class Program
```

File: src/UKHO.S100PermitService.Stubs/Program.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        static void Main()
```
Comment (Nadeem14456) [inline src/UKHO.S100PermitService.Stubs/Program.cs:12]:
```suggestion
        private static void Main()
```

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]
Few minor comments to address.
Rest LGTM.

File: src/UKHO.S100PermitService.Stubs/Program.cs (hunk 2 above)
Comment (shirinbt14926) [reply src/UKHO.S100PermitService.Stubs/Program.cs]: Resolved

File: src/UKHO.S100PermitService.Stubs/Program.cs (hunk 1 above)
Comment (shirinbt14926) [reply src/UKHO.S100PermitService.Stubs/Program.cs]: Resolved

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Review by jrippington — CHANGES_REQUESTED** [review]
Looks good, few minor adjustments, thanks!

File: src/UKHO.S100PermitService.Stubs/Program.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+{
+    internal class Program
+    {
+        private static void Main()
+        {
+            var configuration = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json")
+                .Build();
+
+            var services = new ServiceCollection();
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply src/UKHO.S100PermitService.Stubs/Program.cs:18]: Resolved. Thanks.

File: src/UKHO.S100PermitService.Stubs/Program.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            services.Configure<StubConfiguration>(configuration.GetSection("StubConfiguration"));
+            services.Configure<ShopFacadeApi>(configuration.GetSection("ShopFacadeApi"));
+            services.Configure<ProductKeyServiceApi>(configuration.GetSection("ProductKeyServiceApi"));
+            var serviceProvider = services.BuildServiceProvider();
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply src/UKHO.S100PermitService.Stubs/Program.cs:24]: Resolved. Thanks.

File: src/UKHO.S100PermitService.Stubs/Program.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+            var stubConfig = serviceProvider.GetService<IOptions<StubConfiguration>>()?.Value;
+            var shopFacadeApi = serviceProvider.GetService<IOptions<ShopFacadeApi>>()?.Value;
+            var productKeyServiceApi = serviceProvider.GetService<IOptions<ProductKeyServiceApi>>()?.Value;
+
+            var server = WireMockServer.Start(new WireMockServerSettings
+            {
+                Port = stubConfig?.Port,
+                UseSSL = true
+            });
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply src/UKHO.S100PermitService.Stubs/Program.cs:34]: Resolved. Thanks.

File: src/UKHO.S100PermitService.Stubs/Stubs/ShopFacadeApiStub.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Net;
+using UKHO.S100PermitService.Stubs.Configuration;
+using WireMock.RequestBuilders;
+using WireMock.ResponseBuilders;
+using WireMock.Server;
+
+namespace UKHO.S100PermitService.Stubs.Stubs
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply src/UKHO.S100PermitService.Stubs/Stubs/ShopFacadeApiStub.cs]: Resolved. Thanks.

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

File: src/UKHO.S100PermitService.Stubs/StubConfigurationManager.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿using WireMock.Server;
+
+namespace UKHO.S100PermitService.Stubs
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply src/UKHO.S100PermitService.Stubs/StubConfigurationManager.cs]: Resolved. Thanks.

File: src/UKHO.S100PermitService.Stubs/Configuration/StubConfiguration.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+﻿namespace UKHO.S100PermitService.Stubs.Configuration
+{
+    public class StubConfiguration
```
[the remark this answers is not in the document]
Comment (chaital14503) [reply src/UKHO.S100PermitService.Stubs/Configuration/StubConfiguration.cs]: Resolved. Thanks.

**Pull request merged**